### PR TITLE
fix: cdk graph imported resources and diagram documentation

### PR DESCRIPTION
**Pull request by JeremyJonas**
Imported resources (eg: `s3.Bucket.fromBucketArn()`) were not being references and were not attributed any `cfnType`, which resulted is decoupled and unstylized resources in diagraming.

Additionally the diagram plugin was missing documentation on configurations.

This work started while attempting to reproduce https://github.com/aws/aws-prototyping-sdk/issues/271, which I could not, but that surfaced these gaps in the handling of imported resources.

### CODE REVIEW

File: packages/cdk-graph-plugin-diagram/README.md [hunk 1]
```diff
@@ -61,9 +61,72 @@ This plugin generates diagrams utilizing the [cdk-graph](https://aws.github.io/a
 
 ### Configuration
 
-See [API Documentation](https://aws.github.io/aws-prototyping-sdk/typescript/cdk-graph-plugin-diagram/index.html) for details, and look in [unit tests](https://github.com/aws/aws-prototyping-sdk/tree/mainline/packages/cdk-graph-plugin-diagram/test/graphviz) for more examples.
+See [API Documentation](https://aws.github.io/aws-prototyping-sdk/typescript/cdk-graph-plugin-diagram/index.html#ipluginconfig) for details, and look in [unit tests](https://github.com/aws/aws-prototyping-sdk/tree/mainline/packages/cdk-graph-plugin-diagram/test/graphviz) for more examples.
 
-#### Example Configurations (expand below)
+By default the diagram plugin will generate a single "compact" preset diagram.
+It is capable of create multiple diagrams each with different configurations, as well as defining the defaults to use.
```
Comment (cogwirrel) [inline packages/cdk-graph-plugin-diagram/README.md]: nit: cabable of _creating_

File: packages/cdk-graph/src/core/graph.ts [hunk 2]
```diff
@@ -287,6 +290,31 @@ export namespace Graph {
       this._logicalIdLookup.set(uid, resource.uuid);
     }
 
+    /**
+     * Records arn tokens from imported resources (eg: `s3.Bucket.fromBucketAr()`)
```
Comment (cogwirrel) [inline packages/cdk-graph/src/core/graph.ts]: nit: `fromBucketArn()`

File: packages/cdk-graph/src/core/utils.ts [hunk 3]
```diff
@@ -233,3 +266,96 @@ export function inferFlags(
...
+ */
+export function tokenizeImportArn(value: any): string {
+  return generateConsistentUUID(value, "ImportArn-");
+}
+
+/**
+ * Infers CloudFormation Type for a given import resource.
+ * @param construct {Construct} Import construct such as `s3.Bucket.fromBucketArn()`.
+ * @param constructInfo {ConstructInfo} Construct info like fqn
+ * @returns Returns Cloudformation Type is can infer, otherwise undefined.
```
Comment (cogwirrel) [inline packages/cdk-graph/src/core/utils.ts]: nit: _if_ can infer

**Review by cogwirrel — DISMISSED** [review]
Looks great! Just a few small typos :)

File: packages/cdk-graph-plugin-diagram/README.md [hunk 4]
```diff
@@ -61,9 +61,72 @@ This plugin generates diagrams utilizing the [cdk-graph](https://aws.github.io/a
 
 ### Configuration
 
-See [API Documentation](https://aws.github.io/aws-prototyping-sdk/typescript/cdk-graph-plugin-diagram/index.html) for details, and look in [unit tests](https://github.com/aws/aws-prototyping-sdk/tree/mainline/packages/cdk-graph-plugin-diagram/test/graphviz) for more examples.
+See [API Documentation](https://aws.github.io/aws-prototyping-sdk/typescript/cdk-graph-plugin-diagram/index.html#ipluginconfig) for details, and look in [unit tests](https://github.com/aws/aws-prototyping-sdk/tree/mainline/packages/cdk-graph-plugin-diagram/test/graphviz) for more examples.
```
Comment (agdimech) [inline packages/cdk-graph-plugin-diagram/README.md]: Are we able to also change the example on line 36 to add () at the end of the block as it is currently missing?

File: packages/cdk-graph-plugin-diagram/README.md (hunk 1 above)
Comment (agdimech) [reply packages/cdk-graph-plugin-diagram/README.md]: *capable

**Review by cogwirrel — APPROVED** [review]

File: packages/cdk-graph-plugin-diagram/README.md (hunk 1 above)
Comment (JeremyJonas) [reply packages/cdk-graph-plugin-diagram/README.md]: fixed

File: packages/cdk-graph/src/core/utils.ts (hunk 3 above)
Comment (JeremyJonas) [reply packages/cdk-graph/src/core/utils.ts]: Fixed

File: packages/cdk-graph-plugin-diagram/README.md (hunk 4 above)
Comment (JeremyJonas) [reply packages/cdk-graph-plugin-diagram/README.md]: Fixed

File: packages/cdk-graph/src/core/graph.ts (hunk 2 above)
Comment (JeremyJonas) [reply packages/cdk-graph/src/core/graph.ts]: fixed

**Pull request merged**